File: fem/C++/openacc/auto.def/src/input_cntl.c
/**
 ** INPUT_CNTL
 **/
#include <stdio.h>
#include <stdlib.h>
#include "pfem_util.h"
/** **/
void INPUT_CNTL( char* input )
{
	FILE *fp;
	if( (fp=fopen( input,"r")) == NULL){
		fprintf(stdout,"input file cannot be opened!\n");
		exit(1);
	}
	fscanf(fp,"%s",fname);
	fscanf(fp,"%d",&ITER);
#if   FP ==  32
	fscanf(fp, "%f   %f", &COND, &QVOL);
	fscanf(fp, "%f",      &RESID);
#elif FP ==  64
	fscanf(fp, "%lf %lf", &COND, &QVOL);
	fscanf(fp, "%lf",     &RESID);
#elif FP == 128
	fscanf(fp, "%Lf %Lf", &COND, &QVOL);
	fscanf(fp, "%Lf",     &RESID);
#endif    
	fclose(fp);

	pfemRarray[0]= RESID;
	pfemIarray[0]= ITER;
}

File: fem/F/openacc/auto.def/src/input_cntl.f
!C
!C***
!C*** INPUT_CNTL
!C***
!C
      subroutine INPUT_CNTL( input )
      use pfem_util
      implicit none

      character(*), intent(in) :: input

        open (11,file=input, status='unknown')
        read (11,'(a80)') fname
        read (11,*) ITER
        read (11,*) COND, QVOL
        read (11,*) RESID
        close (11)

      pfemIarray(1)= ITER
      pfemRarray(1)= RESID

      return
      end
